src/main/java/io/unlogged/AgentCommandExecutorImpl.java

```diff
@@ -483,23 +483,104 @@ public AgentCommandResponse injectMocks(AgentCommandRequest agentCommandRequest)
 
         for (String sourceClassName : mocksBySourceClass.keySet()) {
             Object sourceClassInstance = logger.getObjectByClassName(sourceClassName);
+			
+			Class<? extends Object> classObject = null;
             if (sourceClassInstance == null) {
-                // no instance found for this class
-                // nothing to inject mocks to
-                continue;
+
+				// define applicationContext and springBeanFactory
+				if (this.springTestContextManager == null) {
```

Comment:
the block for spring context setup should be refactored to a separate method.

Reply:
Done, refactored the method.